src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -118,6 +119,13 @@ public Conversion ClassifyImplicitConversionFromExpression(BoundExpression sourc
                     }
                 }
             }
+            else if (sourceExpression.GetFunctionType() is { })
+            {
+                if (IsValidFunctionTypeConversionTarget(destination, ref useSiteInfo))
```

Comment:
>if (IsValidFunctionTypeConversionTarget(destination, ref useSiteInfo)) [](http://example.com/codeflow?start=16&length=70)

Should we also check conversions between two Function types?

Reply:
The only code paths that should have to handle conversions between two function types are `BestTypeInferrer` and `MethodTypeInferrer` in cases where "bounds" are merged. Those code paths use `ClassifyImplicitConversionFromTypeOrImplicitFunctionTypeConversion()`.

Reply:
Added handling of conversions between function types.